components/camel-huaweicloud-smn/src/test/java/org/apache/camel/component/huaweicloud/smn/PublishTemplatedMessageTest.java

```diff
@@ -0,0 +1,129 @@
+package org.apache.camel.component.huaweicloud.smn;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.verification.LoggedRequest;
+import org.apache.camel.BindToRegistry;
+import org.apache.camel.Exchange;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.huaweicloud.smn.models.ServiceKeys;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.test.junit4.CamelTestSupport;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.mockito.exceptions.base.MockitoException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
```

Comment:
In general, I would like us to avoid the external mocking as much as possible. Did you consider to mock `SmnClient` instead? Meaning that, create a test SmnClient client that overrides the functionality of  `SmnClient` with your mocked functionality instead. And then add the ability in the endpoint to set the client explicitly. With that you won't need wiremock, your service is mocked by yourself, e.g: https://github.com/apache/camel/blob/master/components/camel-aws-s3/src/test/java/org/apache/camel/component/aws/s3/AmazonS3ClientMock.java and then you can see that in the other tests, the mocked client was supplied

Reply:
Sounds like a better idea. I will get the wiremock removed and instead have a mocked client. Thanks

Reply:
Suggestion has been incorporated into the code. Main codebase has been re-visited to  make it  more testable with above suggestion